desired_state_generator/src/gen_desired_state.py

```diff
@@ -0,0 +1,205 @@
+# Copyright (c) 2023 Robert Bosch GmbH
+#
+# This program and the accompanying materials are made available under the
+# terms of the Apache License, Version 2.0 which is available at
+# https://www.apache.org/licenses/LICENSE-2.0.
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+# WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+# License for the specific language governing permissions and limitations
+# under the License.
+#
+# SPDX-License-Identifier: Apache-2.0
+
+import argparse
+import hashlib
+import json
+import re
+from typing import Any, Dict, List
+
+import requests
+import velocitas_lib
+import velocitas_lib.services
+
+
+def is_uri(path: str) -> bool:
+    """Check if the provided path is a URI.
+
+    Args:
+        path (str): The path to check.
+
+    Returns:
+        bool: True if the path is a URI. False otherwise.
+    """
+    return re.match(r"(\w+)\:\/\/(\w+)", path) is not None
+
+
+def parse_vehicle_signal_interface(config: Dict[str, Any]) -> List[str]:
+    """Parses the vehicle signal interface config.
```

Comment:
Should be imperative, pydocstyle should complain.

```suggestion
    """Parse the vehicle signal interface config.
```